src/frontend/src/optimizer/plan_node/stream_hash_join.rs

```diff
@@ -97,12 +94,32 @@ impl StreamHashJoin {
     pub(super) fn derive_dist(
         left: &Distribution,
         right: &Distribution,
-        l2o_mapping: &ColIndexMapping,
+        logical: &LogicalJoin,
     ) -> Distribution {
         match (left, right) {
             (Distribution::Single, Distribution::Single) => Distribution::Single,
             (Distribution::HashShard(_), Distribution::HashShard(_)) => {
-                l2o_mapping.rewrite_provided_distribution(left)
```

Comment:
What is this part of the changes for?

Reply:
https://github.com/risingwavelabs/risingwave/pull/5344/files/5bf2ab3e46efcb0fb19bba41d05e552ba7a6b8f5#r973925141